core/layout_api.php

```diff
@@ -404,50 +404,63 @@ function layout_login_page_end() {
  * @return null
  */
 function layout_navbar() {
-	$t_logo_url = config_get('logo_url');
-
-	echo '<div id="navbar" class="navbar navbar-default navbar-collapse navbar-fixed-top noprint">';
-	echo '<div id="navbar-container" class="navbar-container">';
-
-	echo '<button id="menu-toggler" type="button" class="navbar-toggle menu-toggler pull-left hidden-lg" data-target="#sidebar">';
-	echo '<span class="sr-only">Toggle sidebar</span>';
-	echo '<span class="icon-bar"></span>';
-	echo '<span class="icon-bar"></span>';
-	echo '<span class="icon-bar"></span>';
-	echo '</button>';
-
-	echo '<div class="navbar-header">';
-	echo '<a href="' . $t_logo_url . '" class="navbar-brand">';
-	echo '<span class="smaller-75"> ';
-	echo string_display_line( config_get('window_title') );
-	echo ' </span>';
-	echo '</a>';
-
-	$t_toggle_class = (OFF == config_get('show_avatar') ? 'navbar-toggle' : 'navbar-toggle-img');
-	echo '<button type="button" class="navbar-toggle ' . $t_toggle_class . ' collapsed pull-right hidden-sm hidden-md hidden-lg" data-toggle="collapse" data-target=".navbar-buttons,.navbar-menu">';
-	echo '<span class="sr-only">Toggle user menu</span>';
-	if (auth_is_user_authenticated()) {
-		layout_navbar_user_avatar();
-	}
-	echo '</button>';
-
-	echo '</div>';
+	$t_logo_url = config_get( 'logo_url' );
+	$t_toggle_class = ( OFF == config_get( 'show_avatar' ) ? 'navbar-toggle' : 'navbar-toggle-img' );
+
+	ob_start();
+	event_signal( 'EVENT_LAYOUT_NAVBAR' );
+	$t_event_navbar_output = ob_get_clean();
+	?>
+	<div id="navbar" class="navbar navbar-default navbar-collapse navbar-fixed-top noprint">
+		<div id="navbar-container" class="navbar-container">
+
+			<button id="menu-toggler" type="button" class="navbar-toggle menu-toggler pull-left hidden-lg" data-target="#sidebar">
+				<span class="sr-only">Toggle sidebar</span>
+				<span class="icon-bar"></span>
+				<span class="icon-bar"></span>
+				<span class="icon-bar"></span>
+			</button>
```

Comment:
Couple of test scenarios that I can think of:
1- Mobile screens
2- Right to left languages